elucidator-examples/c/monte_carlo/monte_carlo.c

```diff
@@ -0,0 +1,166 @@
+ElucidatorStatus setup(SessionHandle * sh, ErrorHandle * eh);
+ElucidatorStatus run_experiment(SessionHandle * sh, ErrorHandle * eh, uint64_t n_steps, uint64_t samples_per_step);
+StepSummary run_experiment_step(uint64_t samples_per_step);
+ElucidatorStatus run_analysis(SessionHandle * sh, ErrorHandle * eh, uint64_t n_steps, uint64_t display_interval);
+StepSummary summarize_bufnodes(BufNode **bufnodes);
+
+int main() {
+    srand(time(NULL));
+
+    SessionHandle * sh = ALLOCATE_HANDLE();
+    ErrorHandle * eh = ALLOCATE_HANDLE();
+    ElucidatorStatus status;
+
+    status = setup(sh, eh);
+    if (status != ELUCIDATOR_OK) {
+        fprintf(stderr, "Setup failed. Exiting.");
+        return 1;
+    }
+    run_experiment(sh, eh, N_STEPS, SAMPLES_PER_STEP);
+    if (status != ELUCIDATOR_OK) {
+        fprintf(stderr, "Experiment failed. Exiting.");
+        return 1;
+    }
+    run_analysis(sh, eh, N_STEPS, DISPLAY_INTERVAL);
+    if (status != ELUCIDATOR_OK) {
+        fprintf(stderr, "Analysis failed. Exiting.");
+        return 1;
+    }
+    free(sh);
+    free(eh);
+}
+
+
+ElucidatorStatus print_error_if_not_ok(ErrorHandle * eh, ElucidatorStatus status) {
+    if ( status != ELUCIDATOR_OK ) {
+        char * msg = get_error_string(eh);
+        fprintf(stderr, "%s\n", msg);
+        free(msg);
+    }
+    return status;
+}
+
+ElucidatorStatus setup(SessionHandle * sh, ErrorHandle * eh) {
+    ElucidatorStatus status;
+    char * designation = "state";
+    char * spec = "hits: u64, misses: u64";
+
+    status = new_session(sh, ELUCIDATOR_RTREE);
+    if (print_error_if_not_ok(eh, status) != ELUCIDATOR_OK) return status;
+    status = add_spec_to_session(designation, spec, sh, eh);
+    if (print_error_if_not_ok(eh, status) != ELUCIDATOR_OK) return status;
+    return status;
+}
+
+ElucidatorStatus run_experiment(SessionHandle * sh, ErrorHandle * eh, uint64_t n_steps, uint64_t samples_per_step) {
+    ElucidatorStatus status;
+    Point a, b;
+    BoundingBox bb;
+
+    for (uint64_t step = 0; step < n_steps; step++) {
+        StepSummary summary = run_experiment_step(samples_per_step);
+        // Upper and lower bounds for experiment region: x, y, z, t
+        Point a = { -1.0, -1.0, -1.0, 0};
+        Point b = {1.0, 1.0, 1.0, step};
+        BoundingBox bb = {a, b};
+        uint8_t *blob = (uint8_t*)&summary;
+        status = insert_metadata_in_session(sh, bb, "state", blob, sizeof(summary), eh);
+        if (print_error_if_not_ok(eh, status) != ELUCIDATOR_OK) return status;
+    }
+    
+    return status;
+}
+
+StepSummary run_experiment_step(uint64_t samples_per_step) {
+    double x, y;
+    uint64_t hits = 0;
+    uint64_t misses = 0;
+    for (uint64_t i = 0; i < samples_per_step; i++) {
+        x = (double)rand() / RAND_MAX;
+        y = (double)rand() / RAND_MAX;
+
+        if (x * x + y * y <= 1) {
+            hits++;
+        }
+        else {
+            misses++;
```

Comment:
Changed post-increment to pre-increment both to match style in the other file and because pre-increment should be favored over post-increment generally where applicable.